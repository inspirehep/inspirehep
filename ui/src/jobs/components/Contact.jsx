import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { List } from 'immutable';
import ContactList from './ContactList';

class Contact extends Component {
  render() {
    const { contactDetails } = this.props;
    return (
      contactDetails && (
        <div>
          <strong>Contact: </strong>
          <ContactList contacts={contactDetails} wrapperClassName="di" />
        </div>
      )
    );
  }
}

Contact.propTypes = {
  contactDetails: PropTypes.instanceOf(List),
};

Contact.defaultProps = {
  contactDetails: null,
};

export default Contact;
