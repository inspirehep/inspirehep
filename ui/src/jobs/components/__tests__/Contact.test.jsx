import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import Contact from '../Contact';

describe('Contact', () => {
  it('renders with contactDetails', () => {
    const contactDetails = fromJS([
      {
        email: 'johndoe@yahoo.com',
        name: 'John',
      },
      {
        email: 'johndoe2@yahoo.com',
        name: 'John2',
      },
    ]);
    const wrapper = shallow(<Contact contactDetails={contactDetails} />);
    expect(wrapper).toMatchSnapshot();
  });
});
