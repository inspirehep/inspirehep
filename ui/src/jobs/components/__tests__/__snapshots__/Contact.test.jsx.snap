// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders with contactDetails 1`] = `
<div>
  <strong>
    Contact: 
  </strong>
  <ContactList
    contacts={
      Immutable.List [
        Immutable.Map {
          "email": "johndoe@yahoo.com",
          "name": "John",
        },
        Immutable.Map {
          "email": "johndoe2@yahoo.com",
          "name": "John2",
        },
      ]
    }
    wrapperClassName="di"
  />
</div>
`;
